src/core/Typing.ml

```diff
@@ -416,11 +417,11 @@ and check_intro cx dlbl params constr tms =
   in
 
   let _ : D.env =
-    List.fold_right2
-      (fun (_, spec) tm tyenv -> check_argument tyenv spec tm)
+    List.fold_left2
+      (fun tyenv (lbl, spec) tm -> check_argument tyenv lbl spec tm)
+      (D.Env.append V.empty_env params)
       (Desc.Constr.specs constr)
       tms
-      (D.Env.append V.empty_env params)
```

Comment:
BTW my perfectly symmetric mind is owning me again; I am confused why I needed a fold_left? Did I reverse something else? (Please note, I never used `List.rev`)